py3status/modules/mysql_slave.py

```diff
@@ -0,0 +1,93 @@
+# -*- coding: utf-8 -*-
+"""
+Display MySQL/MariaDB slave state.
+
+This module displays the number of seconds the slave is behind the master.
+
+Configuration parameters:
+    cache_timeout: refresh cache_timeout for this module (default 5)
+    format_fail: format used for master_servers
+      (default '[\?color=seconds_behind_master {host} is master]')
+    format_ok: format used for slave servers
+      (default '[\?color=seconds_behind_master {host} is {seconds_behind_master}s behind]')
+    host: the host to connect to (default None)
+    passwd: the password to use for the connection (default None)
+    port: the port to connect to (default 3306)
+    query: The sql query to run (default 'show slave status')
+    thresholds: a threshold list of tuples, -1 used for a master hosts
+        (default {'seconds_behind_master':
+            [(0, 'deepskyblue'), (100, 'good'), (300, 'degraded'), (600, 'bad')]
+        })
+    user: the user used for the connection (default None)
+
+Requires:
+    MySQLdb: python package for python
+
+@author cereal2nd
+@license BSD
+
+SAMPLE OUTPUT
+{'full_text': 'test is 0s behind'}
+{'full_text': 'test is 509s behind'}
+{'full_text': 'test is master'}
+"""
+
+from __future__ import absolute_import
+from MySQLdb import connect
+
+
+class Py3status:
+    """
+    """
+    # available configuration parameters
+    cache_timeout = 5
+    format_fail = '[\?color=seconds_behind_master {host} is master]'
+    format_ok = '[\?color=seconds_behind_master {host} is {seconds_behind_master}s behind]'
```

Comment:
Your new `format` should be this:
```
    format = '\?color=seconds_behind_master {host} is [{seconds_behind_master}s behind|master]'
```
Just kill the `form` lines... leaving `self.format, data` in the return.

Reply:
but how do we return something in case we are master?

we need to return a different format in that case ...

Reply:
There are many ways to do this.
* `'{does_not_exist}|Master!'`
* `'STUFFS HERE [{does_not_exist}|Master!]'`
* `'\?if=!does_not_exist Master Here, Slave here'`

The issue is that the documentation for `formatters` need to be written / exposed so more people can understand this stuffs better.

There is `\?not_zero`, `?if=`, `[{placeholder}]`, `\?color=`, etc...

Reply:
the formatters you provide do not work for a case where seconds_behind_master does not exists,
in that case the 'master' is not displayed ....

```
{'full_text': u'localhost is ', 'cached_until': 1521040851.0}
```
```
{'color': '#00bfff', 'full_text': u'localhost is 0s behind', 'cached_until': 1521040872.0}
```

Reply:
This is a returning value `0` ?

Reply:
no the value does not exists in case of a master

Reply:
```
'[\?if=seconds_behind_master {seconds_behind_master}s behind|master]') 
```

works

Reply:
This one. :-) 
```
    format = ('\?color=seconds_behind_master {host} is '
              '[{seconds_behind_master}s behind|\?show master]')
```

Reply:
This is because it's not a placeholder so we use `\?show` instead.

Reply:
Just saw your example. It's better. use that.